beacon-chain/blockchain/head.go

```diff
@@ -158,12 +158,16 @@ func (s *Service) saveHead(ctx context.Context, newHeadRoot [32]byte, headBlock
 
 	// Forward an event capturing a new chain head over a common event feed
 	// done in a goroutine to avoid blocking the critical runtime main routine.
-	go func() {
-		if err := s.notifyNewHeadEvent(ctx, newHeadSlot, headState, newStateRoot[:], newHeadRoot[:]); err != nil {
-			log.WithError(err).Error("Could not notify event feed of new chain head")
-		}
-	}()
-
+	optimistic, err := s.IsOptimistic(ctx)
```

Comment:
moving the check here so that the lock to forkchoice required by `IsOptimistic` is not acquired on a go routine.